Test.c

```diff
@@ -162,6 +164,52 @@ Test(dci1Test, dci1_calculateShiftOriginTest)
 	{
 		dci1_offsetArray[dci1_bitmap] = possibleLengthBitsRBG[i];
 		testVal = dci1_calculateShiftOrigin(dci1_offsetArray);
-		cr_expect(testVal = correctValues[i], "Value returned from dci1_calculateShiftOrigin [%d] test loop are incorrect", i);
+		cr_expect(testVal = correctValues[i], "Value returned from dci1_calculateShiftOrigin "
+				"[%d] test loop are incorrect", i);
+	}
+}
+
+Test(endToEndTests, dci1_endToEndTest)
+{
+#define TEST_ARR_LEN 4
+	bandwidth_t dci_bandwidth[TEST_ARR_LEN] = {5, 4, 3, 2};
+	uint64_t inputArguments[TEST_ARR_LEN] =  {0x0420005b74, 0x272006dd00, 0x34e01b7c, 0x47a3d7e0};
+	uint32_t properReadVal[TEST_ARR_LEN][DCI1_NUMBER_PARAM] = {{0, 1081345, 13, 5, 1, 2, 2},
+			{0, 160256, 13, 5, 1, 2, 2}, {0, 54144, 13, 5, 1, 3, 2},{0, 4584, 30, 5, 1, 3, 3}};
+	uint32_t properIndexVal[][7] = {{3,4,9,24}, {5,1,4,5,6,9},
+			{6, 1, 2, 4, 7, 8, 9}, {6, 0, 4, 5, 6, 7, 9}};
+	uint32_t reverseIndexValArray;
+
+	for (uint32_t i = 0; i < TEST_ARR_LEN; i++)
+	{
+		uint32_t dci1_offsetArray[DCI1_NUMBER_PARAM] = {RA, BITMAP_LEN, MCS, HARQ, NDI, RV, TPC};
+
+		dci1_offsetArray[dci1_bitmap] = dci1_lengthOfBitmapViaBandwidth(dci_bandwidth[i]);
```

Comment:
Name "dci1_bitmap" in enum type "dci1_OutputParameters" is unclear - when someone doesn't know exactly your code it will look like you are using some bitmap for indexing an array - which seems bad, please use better name for it